inference-engine/src/mkldnn_plugin/ngraph_transformations/split_conv_concat.cpp

```diff
@@ -0,0 +1,121 @@
+    if (!split) {
+        return false;
+    }
+    const auto axisNode = ngraph::get_constant_from_source(split->input_value(1));
+    if (!axisNode) {
+        return false;
+    }
+    auto axis = axisNode->cast_vector<int64_t>()[0];
+    if (axis < 0) {
+        axis += split->get_input_shape(0).size();
+    }
+    const auto splitOutputsNum = split->get_output_size();
+    if (axis != 1 || splitOutputsNum < 2) {
+        return false;
+    }
+
+    // get first convolution attribute, with which we will compare other convolutions
+    auto splitOutputs = split->get_output_target_inputs(0);
+    if (splitOutputs.size() != 1) {
+        return false;
+    }
+    const auto convRef = std::dynamic_pointer_cast<ngraph::opset1::Convolution>(splitOutputs.begin()->get_node()->shared_from_this());
+    if (!convRef || !std::dynamic_pointer_cast<ngraph::opset1::Constant>(convRef->input_value(1).get_node_shared_ptr()) || convRef->get_output_size() != 1) {
+        return false;
+    }
+    const auto stridesRef = convRef->get_strides();
+    const auto padsBeginRef = convRef->get_pads_begin();
+    const auto padsEndRef = convRef->get_pads_end();
+    const auto dilationsRef = convRef->get_dilations();
+    const auto autoPadRef = convRef->get_auto_pad();
+
+    const auto convOutputs = convRef->get_output_target_inputs(0);
+    const auto concatRef = std::dynamic_pointer_cast<ngraph::opset1::Concat>(convOutputs.begin()->get_node()->shared_from_this());
+    if (!concatRef) {
+        return false;
+    }
+    auto concatAxis = concatRef->get_axis();
+    if (axis < 0) {
+        concatAxis += concatRef->get_output_shape(0).size();
+    }
+    if (axis != 1) {
+        return false;
+    }
+
+    for (size_t i = 1; i < splitOutputsNum; i++) {
+        auto splitOutputs = split->get_output_target_inputs(i);
+        if (splitOutputs.size() != 1) {
+            return false;
+        }
+        const auto conv = std::dynamic_pointer_cast<ngraph::opset1::Convolution>(splitOutputs.begin()->get_node()->shared_from_this());
+        if (!conv || !std::dynamic_pointer_cast<ngraph::opset1::Constant>(conv->input_value(1).get_node_shared_ptr()) || conv->get_output_size() != 1) {
+            return false;
+        }
+        if (conv->get_strides() != stridesRef || conv->get_pads_begin() != padsBeginRef || conv->get_pads_end() != padsEndRef ||
+                conv->get_dilations() != dilationsRef || conv->get_auto_pad() != autoPadRef) {
+            return false;
+        }
+        const auto convOutputs = conv->get_output_target_inputs(0);
+        const auto concat = std::dynamic_pointer_cast<ngraph::opset1::Concat>(convOutputs.begin()->get_node()->shared_from_this());
+        if (!concat || concatRef != concat) {
+            return false;
+        }
+    }
+
+    return true;
+}
+
+MKLDNNPlugin::SplitConvConcatPattern::SplitConvConcatPattern() {
+    auto split = ngraph::pattern::wrap_type<ngraph::opset1::Split>();
```

Comment:
Could you please match Concat instead of Split? It is better not to use passes down the graph.

Reply:
fixed